pkgs/development/python-modules/babeltrace2/default.nix

```diff
@@ -0,0 +1,17 @@
+{ toPythonModule
+, python
+, babeltrace2
+, swig
+}:
+
+toPythonModule (babeltrace2.overrideAttrs ({ nativeBuildInputs ? [ ], configureFlags ? [ ], ... }: {
```

Comment:
IMO the interface you used in the first commits is not worth it if eventually you use `overrideAttrs`.

Reply:
The interface is for covering the differences between v1 and v2. Overriding is used for changes in Python package builds. If we extend the interface from the first commit to also allow conditional building of Python packages (e.g. by adding `enablePython` as you suggest bellow), this would complicate the expression a lot. I find it more manageable to use overrides here.

Reply:
> The interface is for covering the differences between v1 and v2. Overriding is used for changes in Python package builds. If we extend the interface from the first commit to also allow conditional building of Python packages (e.g. by adding `enablePython` as you suggest bellow), this would complicate the expression a lot. I find it more manageable to use overrides here.

Personally it feels more complicated to me to use 2 different overriding schemes, and scattering the logic between different files. After all both derivations are built from the same source. I also find it very confusing the `nativeBuildInputs ? []` and `configureFlags ? []` function... What is the value given by reading these? 

If you hadn't had used a function with two attribute sets as arguments it would have been at least more readable.

Reply:
> I also find it very confusing the `nativeBuildInputs ? []` and `configureFlags ? []` function...

Yeah, I thought that this is a bit awkward. I'll try to come up with a cleaner solution. Perhaps splitting v1 and v2 as you suggest.